devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy

```diff
@@ -0,0 +1,130 @@
+import physicalgraph.zigbee.clusters.iaszone.ZoneStatus
+metadata {
+    definition (name: "GatorSystem HomeWatcher", namespace: "GS", author: "GS_coder000") {
+    capability "Motion Sensor"
+    capability "Battery"
+    capability "Contact Sensor"
+    capability "Presence Sensor"   
+        
+    // Raw Description 08 0104 0402 00 02 0000 0500 01 0502
+    fingerprint deviceJoinName: "GatorSystem Multipurpose Sensor", manufacturer: "GatorSystem", model: "GSHW01"
+    }
+}
+
+def parse(String description) {
+    log.debug "${device.displayName} description: $description"
+    Map map = [:]
+    if (description?.startsWith('catchall:')) { //raw commands that smartthings does not or cannot interpret
+        map = zigbee.parseDescriptionAsMap(description)   
+    } else if (description?.startsWith('read attr -')) {
+        map = zigbee.parseDescriptionAsMap(description)
+    } else if (description?.startsWith('zone status')) {
+        map = parseIasMessage(description)
+    }
+    log.debug "Parse returned map $map"
+    if (map != null) {    
```

Comment:
Wouldn't it look better if You returned createEvent(map) here and added else {return null} statement ?

Reply:
I rethinked that,  please get rid of that last 'else {return null}'  statement, it's redundant.